src/Shared/EndpointMetadataPopulator.cs

```diff
@@ -12,6 +12,7 @@
 namespace Microsoft.AspNetCore.Http;
 
 [UnconditionalSuppressMessage("Trimmer", "IL2060", Justification = "Trimmer warnings are presented in RequestDelegateFactory.")]
+[UnconditionalSuppressMessage("AOT", "IL3050", Justification = "AOT warnings are presented in RequestDelegateFactory.")]
```

Comment:
These shouldn't be suppressed. Instead this class should be marked with `RequiresUnreferenecedCode` and `RequiresDynamicCode`. That way if any new code starts calling it, they will get warnings as well.

Reply:
Looks like this is also called from

https://github.com/dotnet/aspnetcore/blob/a8813d4bcbf2f940cd3e159d266fdd039a390922/src/Mvc/Mvc.Core/src/Routing/ActionEndpointFactory.cs#L370-L373

This is why suppressions like this aren't valid.